Framework/API/src/ScriptRepository.cpp

```diff
@@ -8,6 +8,10 @@
 #ifdef _WIN32
 #define snprintf _snprintf
 #endif
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation"
```

Comment:
Is this warning not fixable rather than ignoring it?

Reply:
This is a conflict between GCC 9.3 and Cppcheck i.e. GCC is claiming there will be a format truncation, if a larger buffer is not given on lines 36 and 53. If a larger buffer is given then CppCheck says the buffer is too large and is a security flaw.

Reply:
I would tend to believe the compiler here and I think we should look at the code and suppress a cppcheck warning if we need to.

Reply:
I've had a look at the exception defined here. I think we can simplify the whole class by first removing the default constructor and also the second constructor accepting an `int` as the first parameter - neither are used anywhere and I don't think are that useful in themselves.

We can then avoid having to suppress this warning by rewriting the final constructor to not use `snprintf` and just use string concatenation and `std::to_string`.